swiftgen config init (#694)
SwiftGen/SwiftGen: Documentation/ConfigFile.md

# SwiftGen Configuration File

SwiftGen expects you to have a `swiftgen.yml` configuration file at the root of your repository, to specify which files to parse, what template to use, and where to write the generated files.

## Creating a Config file

To create your configuration file, you can use the **`swiftgen config init`** command.

This will create (and open) an example configuration file that you can use for inspiration, with some typical entries commented out. All you have to do is to then uncomment the parts of the config that you need, adapt them to your project, and/or copy&paste the examples to add configuration for additional parsers.

> You can of course also create the `swiftgen.yml` configuration file manually from scratch, using the text editor of your choice. The file is just regular a YAML _(note: JSON, being a subset of YAML, is also supported)_

Reviewer: I think that the last sentence is missing some bits.

Author: It isn't really, but I can try to elaborate.

```suggestion
You can of course also create the `swiftgen.yml` configuration file manually from scratch, using the text editor of your choice. The file just has to be in YAML format _(note: JSON, being a subset of YAML, is also supported)_ and provide the keys described below.
```

Reviewer: Yeah that's better. The initial "The file is just regular a YAML" was just incomplete I think.

| Subkey | Type | Description |
|--------|------|-------------|
| `inputs` | Path or Array of Paths | The file(s)/dir(s) to parse (e.g. the path to your assets catalog for the `xcassets` command, or your `.lproj` or `Localizable.strings` file for the `strings` command, etc). |
| `options` | Dictionary | Any optional list of settings for the parser, to change its behaviour. See the commands' specific documentation for available options. |

> | `filter` | Regular Expression | The regular expression to apply to each input path; only paths matching the given filter will be used. Filters are applied to actual (relative) paths, not just the filename. Note that each command has a default built-in filter (based on expected file extensions), so you generally don't need to specify that subkey in your config file; but it is useful if you want to override it with a custom filter. |

Reviewer: ~that~ this subkey

Author: ```suggestion
| `filter` | Regular Expression | The regular expression to apply to each input path; only paths matching the given filter will be used. Filters are applied to actual (relative) paths, not just the filename. Note that each command has a default built-in filter (based on expected file extensions), so you generally don't need to specify this subkey in your config file; but it is useful if you want to override it with a custom filter. |
```

| `templateName` | String | The name of the template to use. If you provide a value for this, you shouldn't also provide a value for `templatePath`. |
| `templatePath` | Path | The path to the template to use. If you provide a value for this, you shouldn't also provide a value for `templateName`. |
| `output` | Path | The path of the output file to generate. _(Note: Any intermediate directory up to this file must already exist.)_ |

> | `params` | Dictionary | Any optional parameter you want to pass to the template. See the template documentation in GitHub (or using `swiftgen template doc` command) to know the list of available params each template supports. |

Reviewer: ~to know~ for the list of available ...

Author: ```suggestion
| `params` | Dictionary | Any optional parameter you want to pass to the template. See the template documentation in GitHub (or using `swiftgen template doc` command) for the list of available params each template supports. |
```
